tests/test_utils.go

```diff
@@ -213,6 +213,18 @@ func PrepareTestEnv(t testing.TB, skip ...int) func() {
 		}
 	}
 
+	// prepare attachments directory and files
+	assert.NoError(t, storage.Clean(storage.Attachments))
+
+	s, err := storage.NewStorage(setting.LocalStorageType, &setting.Storage{
+		Path: filepath.Join(filepath.Dir(setting.AppPath), "tests", "testdata", "data", "attachments"),
+	})
+	assert.NoError(t, err)
+	assert.NoError(t, s.IterateObjects("", func(p string, obj storage.Object) error {
+		_, err = storage.Copy(storage.Attachments, p, s, p)
+		return err
+	}))
```

Comment:
Does every test need such preparation "copy"?

For more than 99% tests, this "copy" just wastes time.

Let the test case prepare the data when it really needs it?

Reply:
Done.